src/compiler/scala/tools/nsc/Settings.scala

```diff
@@ -12,11 +12,12 @@
 
 package scala.tools.nsc
 
-import settings.MutableSettings
+import settings.{DefaultPathFactory, MutableSettings, PathFactory}
 
 /** A compatibility stub.
  */
-class Settings(errorFn: String => Unit) extends MutableSettings(errorFn) {
+class Settings(errorFn: String => Unit, pathFactory: PathFactory) extends MutableSettings(errorFn, pathFactory) {
```

Comment:
Could make sense to obtain the `PathFactory` reflectively, from a compiler flag, similar to [`-Xreporter`](https://github.com/scala/scala/blob/37416816847468c91adbd0c064436cf1ec4a2c77/src/compiler/scala/tools/nsc/settings/ScalaSettings.scala#L136)? For example with the current zinc compiler interface, the new functionality could not be used.

Reply:
There is a bit of a chicken and egg problem, in that we need the path factory in place before we start parsing the `Settings`. That's why I thought a constructor parameter was safest.